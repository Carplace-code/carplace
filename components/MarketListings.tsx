/* eslint-disable */

interface MarketListingsProps {
  listings: any[];
  selectedId: string;
  onSelect: (listing: any) => void;
}

export default function MarketListings({ listings, selectedId, onSelect }: MarketListingsProps) {
  return (
    <div className="rounded-md border bg-white p-6">
      <h2 className="mb-4 text-lg font-semibold">Available On</h2>
      <div className="grid grid-cols-1 gap-4 text-sm md:grid-cols-3">
        {listings.map((item) => {
          const isSelected = item.id === selectedId;
          return (
            <div key={item.id} className={`space-y-2 rounded border p-4 ${isSelected ? "border-blue-500" : ""}`}>
              <p className="font-medium">{item.source?.name}</p>
              <p className="font-bold text-blue-600">CLP ${Number(item.price).toLocaleString()}</p>
              <p className="text-gray-600">Seller: {item.seller?.name}</p>
              <p className="text-gray-600">Location: {item.location}</p>
              <p className="text-xs text-gray-500">Updated: {new Date(item.scrapedAt).toLocaleString()}</p>
              <button
                onClick={() => onSelect(item)}
                className="mt-2 w-full rounded-md border py-1 text-sm hover:bg-gray-100"
              >
                {isSelected ? "Selected" : "View Listing"}
              </button>
            </div>
          );
        })}
      </div>
    </div>
  );
}
