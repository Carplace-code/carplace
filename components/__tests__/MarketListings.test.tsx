import { render } from "@testing-library/react";
import { describe, it } from "vitest";

import MarketListings from "../MarketListings";

describe("MarketListings", () => {
  it("renders correctly and matches snapshot", () => {
    const { container } = render(<MarketListings />);
    expect(container).toMatchSnapshot();
  });
});
