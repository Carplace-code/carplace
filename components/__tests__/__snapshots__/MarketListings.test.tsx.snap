// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MarketListings > renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="rounded-md border bg-white p-6"
  >
    <h2
      class="mb-4 text-lg font-semibold"
    >
      Available On
    </h2>
    <div
      class="grid grid-cols-1 gap-4 text-sm md:grid-cols-3"
    >
      <div
        class="space-y-2 rounded border p-4"
      >
        <p
          class="font-medium"
        >
          Yapo
        </p>
        <p
          class="font-bold text-blue-600"
        >
          $35,990,000
        </p>
        <p
          class="text-gray-600"
        >
          Seller: 
          Juan Pérez
        </p>
        <p
          class="text-gray-600"
        >
          Location: 
          Santiago
        </p>
        <p
          class="text-xs text-gray-500"
        >
          3 hours ago
        </p>
        <button
          class="mt-2 w-full rounded-md border py-1 text-sm hover:bg-gray-100"
          type="button"
        >
          View Original
        </button>
      </div>
      <div
        class="space-y-2 rounded border p-4"
      >
        <p
          class="font-medium"
        >
          Facebook
        </p>
        <p
          class="font-bold text-blue-600"
        >
          $36,700,000
        </p>
        <p
          class="text-gray-600"
        >
          Seller: 
          María González
        </p>
        <p
          class="text-gray-600"
        >
          Location: 
          Temuco
        </p>
        <p
          class="text-xs text-gray-500"
        >
          6 hours ago
        </p>
        <button
          class="mt-2 w-full rounded-md border py-1 text-sm hover:bg-gray-100"
          type="button"
        >
          View Original
        </button>
      </div>
      <div
        class="space-y-2 rounded border p-4"
      >
        <p
          class="font-medium"
        >
          Chileautos
        </p>
        <p
          class="font-bold text-blue-600"
        >
          $39,000,000
        </p>
        <p
          class="text-gray-600"
        >
          Seller: 
          Auto Import SPA
        </p>
        <p
          class="text-gray-600"
        >
          Location: 
          Santiago
        </p>
        <p
          class="text-xs text-gray-500"
        >
          5 hours ago
        </p>
        <button
          class="mt-2 w-full rounded-md border py-1 text-sm hover:bg-gray-100"
          type="button"
        >
          View Original
        </button>
      </div>
    </div>
  </div>
</div>
`;
